File: frontend/src/models/company.ts
import type { Contact } from './contact';
import type { Lead } from './lead';
import type { Deal } from './deal';
import type { CRMTask } from './crmTask';

export type CompanySize = 'micro' | 'small' | 'medium' | 'large' | 'enterprise';
export type CompanyStatus = 'active' | 'inactive' | 'prospect' | 'customer';

export interface Company {
	id: number;
	public_id: string;
	name: string;
	industry?: string;
	company_size?: CompanySize;
	website?: string;
	phone?: string;
	email?: string;
	status: CompanyStatus;
	address?: Record<string, any>;
	social_media?: Record<string, any>;
	custom_fields?: Record<string, any>;
	contacts?: Contact[];
	leads?: Lead[];
	deals?: Deal[];
	tasks?: CRMTask[];
	crm_activities?: any[];
	created_at: string;
	updated_at: string;
}

export interface CompanyCreate extends Omit<Company, 'id' | 'public_id' | 'created_at' | 'updated_at'> { }
export interface CompanyUpdate extends Partial<CompanyCreate> { }

export interface CompanyStats {
	total: number;
	by_status: Record<string, number>;
	top_industries: { industry: string; count: number }[];
}

export interface CompanyPaginatedResponse {
	items: Company[];
	total: number;
}

File: backend/app/schemas/company.py
"""Company Schemas"""

from typing import Optional, Dict, Any
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, ConfigDict, EmailStr, Field
from app.models.company import CompanySize, CompanyStatus
from app.schemas.contact import ContactRead


class CompanyBase(BaseModel):
    name: str = Field(..., max_length=255)
    industry: Optional[str] = Field(None, max_length=100)
    company_size: Optional[CompanySize] = None
    website: Optional[str] = Field(None, max_length=255)
    phone: Optional[str] = Field(None, max_length=50)
    email: Optional[EmailStr] = None
    status: CompanyStatus = CompanyStatus.PROSPECT
    address: Optional[Dict[str, Any]] = None
    social_media: Optional[Dict[str, Any]] = None
    custom_fields: Optional[Dict[str, Any]] = None


class CompanyCreate(CompanyBase):
    pass


class CompanyUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=255)
    industry: Optional[str] = Field(None, max_length=100)
    company_size: Optional[CompanySize] = None
    website: Optional[str] = Field(None, max_length=255)
    phone: Optional[str] = Field(None, max_length=50)
    email: Optional[EmailStr] = None
    status: Optional[CompanyStatus] = None
    address: Optional[Dict[str, Any]] = None
    social_media: Optional[Dict[str, Any]] = None
    custom_fields: Optional[Dict[str, Any]] = None


class CompanyRead(CompanyBase):
    id: int
    public_id: UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CompanyWithContacts(CompanyRead):
    contacts: list[ContactRead] = []


class CompanyStats(BaseModel):
    total: int
    by_status: Dict[str, int]
    top_industries: list[Dict[str, Any]]


class CompanyListResponse(BaseModel):
    items: list[CompanyRead]
    total: int
